presto-main/src/main/java/io/prestosql/execution/QueryManagerConfig.java

```diff
@@ -60,6 +63,7 @@
     private Duration queryMaxRunTime = new Duration(100, TimeUnit.DAYS);
     private Duration queryMaxExecutionTime = new Duration(100, TimeUnit.DAYS);
     private Duration queryMaxCpuTime = new Duration(1_000_000_000, TimeUnit.DAYS);
+    private DataSize maxQueryDataSize = new DataSize(100, TERABYTE);
```

Comment:
We should make this optional or nullable by default, so that users have to opt-in to having a limit.

Reply:
Let’s call it `queryMaxDataSize` to be consistent with the others.

Reply:
Done